embassy-nxp/src/usart/lpc55.rs

```diff
@@ -0,0 +1,854 @@
+            Error::Framing => ErrorKind::InvalidData,
+            Error::Break => ErrorKind::Interrupted,
+            Error::Noise => ErrorKind::Other,
+        }
+    }
+}
+/// Word length.
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub enum DataBits {
+    /// 7 bits.
+    #[doc = "7 bit Data length."]
+    DataBits7,
+    /// 8 bits.
+    #[doc = "8 bit Data length."]
+    DataBits8,
+    /// 9 bits.
+    #[doc = "9 bit data length. The 9th bit is commonly used for addressing in multidrop mode."]
+    DataBits9,
+}
+
+impl DataBits {
+    fn bits(&self) -> u8 {
+        match self {
+            Self::DataBits7 => 0b00,
+            Self::DataBits8 => 0b01,
+            Self::DataBits9 => 0b10,
+        }
+    }
+}
+
+/// Parity bit.
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub enum Parity {
+    /// No parity.
+    #[doc = "No parity."]
+    ParityNone,
+    /// Even parity.
+    #[doc = "Even parity."]
+    ParityEven,
+    /// Odd parity.
+    #[doc = "Odd parity."]
+    ParityOdd,
+}
+
+impl Parity {
+    fn bits(&self) -> u8 {
+        match self {
+            Self::ParityNone => 0b00,
+            Self::ParityEven => 0b10,
+            Self::ParityOdd => 0b11,
+        }
+    }
+}
+
+/// Stop bits.
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub enum StopBits {
+    #[doc = "1 stop bit."]
+    STOP1,
+    #[doc = "2 stop bits. This setting should only be used for asynchronous communication."]
+    STOP2,
+}
+
+impl StopBits {
+    fn bits(&self) -> bool {
+        return match self {
+            Self::STOP1 => false,
+            Self::STOP2 => true,
+        };
+    }
+}
+
+/// UART config.
+#[non_exhaustive]
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub struct Config {
+    /// Baud rate.
+    pub baudrate: u32,
+    /// Word length.
+    pub data_bits: DataBits,
+    /// Stop bits.
+    pub stop_bits: StopBits,
+    /// Parity bit.
+    pub parity: Parity,
+    /// Invert the tx pin output
+    pub invert_tx: bool,
+    /// Invert the rx pin input
+    pub invert_rx: bool,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            baudrate: 9600,
+            data_bits: DataBits::DataBits8,
+            stop_bits: StopBits::STOP1,
+            parity: Parity::ParityNone,
+            invert_rx: false,
+            invert_tx: false,
+        }
+    }
+}
+
+pub struct Usart<'d, T: Instance, M: Mode> {
+    tx: UsartTx<'d, T, M>,
+    rx: UsartRx<'d, T, M>,
+}
+
+pub struct UsartTx<'d, T: Instance, M: Mode> {
+    phantom: PhantomData<(&'d (), T, M)>,
+}
+
+pub struct UsartRx<'d, T: Instance, M: Mode> {
+    phantom: PhantomData<(&'d(), T, M)>,
+}
+
+impl<'d, T: Instance, M: Mode> UsartTx<'d, T, M> {
+    pub fn new(_usart: Peri<'d, T>, tx: Peri<'d, impl TxPin<T>>, config: Config) -> Self {
+        Usart::<T, M>::init( Some(tx.into()), None, config);
+        Self::new_inner()
+    }
+
+    fn new_inner() -> Self {
+        Self {
+            phantom: PhantomData,
+        }
+    }
```

Comment:
Unless `new_inner` is used somewhere else, or could be useful for us in the future, I would just put it directly into `new`
```suggestion
    pub fn new(_usart: Peri<'d, T>, tx: Peri<'d, impl TxPin<T>>, config: Config) -> Self {
        Usart::<T, M>::init( Some(tx.into()), None, config);
        
        Self {
            phantom: PhantomData,
        }
```